libbindgen/src/ir/enum_ty.rs

```diff
@@ -49,8 +49,8 @@ impl Enum {
         }
 
         let declaration = ty.declaration().canonical();
-        let repr = Item::from_ty(&declaration.enum_type(), None, None, ctx)
-            .ok();
+        let et = &declaration.enum_type().expect("This should be an enum since we checked above!");
```

Comment:
@emilio thanks for the input. Question: currently, Item::from_ty (and Item::from_ty_with_id) take a &clang::Type. Therefore, do you suggest that I:

* perform a check in Enum::from_ty to see if declaration.enum_type() returns None, and circuit-break if so, or:
* update Item::from_ty and Item::from_ty_with_id so that they take an Option<Type>, and have them do the validation?

My thinking is that it's the first option, but let me know what you think.